### ISSUE TITLE
MidnightControls

### ISSUE THREAD

**Issue by RaptaG** [Fabulously-Optimized/fabulously-optimized#330]
### Mod name

MidnightControls

### Curseforge link

https://www.curseforge.com/minecraft/mc-mods/midnightcontrols

### Other links

Modrinth: https://modrinth.com/mod/midnightcontrols
 
GitHub: https://github.com/TeamMidnightDust/MidnightControls

### What it does

Provides gamepad support!

### Why should it be in the modpack

It offers MCBE parity, many people when moving from MCBE to Java miss the ability to use a gamepad and using methods with  eg. Steam are very complicated.

### Why shouldn't it be in the modpack

~~It requires the user to add by themselves the gamepad mappings, using an [external app](https://generalarcade.com/gamepadtool/), and some of its extra features can be considered as cheats, so I made [an issue](https://github.com/TeamMidnightDust/MidnightControls/issues/21) for that.~~<br />The settings may be a little confusing.<br />Also, usually, people don't use a controller to play the Java Edition of the game.

### Categories

New feature, Works like Bedrock Edition

### Additional details

It's a fork of [LamdaControls](https://www.curseforge.com/minecraft/mc-mods/lambdacontrols), which is no longer maintained. Also, it's no longer [in beta](https://github.com/TeamMidnightDust/MidnightControls/releases/tag/v1.0.0).<br />
https://github.com/TeamMidnightDust/MidnightControls/issues/13<br />
It should not be considered as a cheat mod ([why](https://github.com/TeamMidnightDust/MidnightControls/issues/21))

**Comment by DioEgizio** [comment]
> It requires the user to add by themselves the gamepad mappings, using an external app, and the settings may be a little confusing.

It can actually download mappings automatically

**Comment by DioEgizio** [comment]
https://github.com/TeamMidnightDust/MidnightControls/commit/cac23d12a1e716cd729f7adeb214d3d695305af2

**Comment by Madis0** [comment]
I want to see comments of people who actually care about this.

**Comment by Kichura** [comment]
I personally care about this because it opens up the accessibility to other people rather than being limited to keyboard & mice - and i feel like without this; wouldn't be as fun to play when game controller fans need such mods.

**Comment by DioEgizio** [comment]
I care about this too, it's also very nice for Steam Deck users (even though I don't have a steam deck)

**Comment by Madis0** [comment]
Okay, let me rephrase this: I want to hear from people who _actively use this (or an equivalent mod)_, not those who just think it is a good idea.

**Comment by Imzxhir** [comment]
I've use both MidnightControls and Controllable (Forge). In my opinion, I think it's a good mod to add because it opens the modpack up to a variety of users on different platforms like the steam deck for instance. It might need some configurartion with the controller options tab but after that it's fine and good to use.

**Comment by Jamelos** [comment]
> I've use both MidnightControls and Controllable (Forge). In my opinion, I think it's a good mod to add because it opens the modpack up to a variety of users on different platforms like the steam deck for instance. It might need some configurartion with the 'General Options' and 'Controller Options' tab but after that it's fine and good to use.

Exactly me.

**Comment by Big-ed69** [comment]
I used lambda controls while it was supported and now use midnight controls instead. Love using it and often just lie down with my laptop and play with my dualshock 4.

**Comment by RaptaG** [comment]
Allow me to add a report too, I've been using MidnightControls for a while now and it is amazing!

**Comment by Madis0** [comment]
Fine, I believe you that there is a demand for this.

Now, the question for those who use it - what do you think would be sensible defaults for this mod in FO to prevent cheating and ease the initial setup? What setup is needed to get this working in the first place?

Note that I have not tried this mod yet and I do not have a controller anyway.

**Comment by Jamelos** [comment]
> Fine, I believe you that there is a demand for this.
> 
> Now, the question for those who use it - what do you think would be sensible defaults for this mod in FO to prevent cheating and to ease the initial setup? What setup is needed to get this working in the first place?
> 
> Note that I have not tried this mod yet and I do not have a controller anyway.

The mod already comes in a way thats not cheating and is equal to the console. If you want, you can enable the cheat features yourself. It functions very simply, put controller in, it detects it and enters Controller Mode. Click the keyboard or disconnect the controller, and it will switch to Keyboard mode. That simple. The mod doesn't affect performance neither show a big not vanilla like screen. Thats it. Just put the mod and bam. No more things, its already ready.

**Comment by RaptaG** [comment]
The new version is out, with bug fixes, modded controller support, auto-adapting controller icons and MidnightControlsExtras which has the “cheat” features.

**Comment by Madis0** [comment]
I tried adding it to FO 4.0.0-beta.5, but it made mining blocks impossible with the left mouse key. Cannot reproduce with a clean profile of only that mod, so it must be something in FO that is conflicting... 🤔

**Comment by RaptaG** [comment]
> (I didn't report it to the mod dev as they have enough unfixed bugs atm, we should figure the root cause out first.)

Somebody did though 🤔:
https://github.com/TeamMidnightDust/MidnightControls/issues/45

**Comment by Madis0** [comment]
Added in 3.12.0 and 4.0.0-beta.6.
